src/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.cs

```diff
@@ -106,7 +107,7 @@ public void SetNext(BufferSegment segment)
 
             while (segment.Next != null)
             {
-                segment.NextSegment.RunningIndex = segment.RunningIndex + segment.Length;
+                segment.NextSegment!.RunningIndex = segment.RunningIndex + segment.Length;
```

Comment:
@davidfowl, is it correct that this is accessing NextSegment but the while above is using Next?  It's not clear to me what the difference is.

Reply:
Seems at least better assert it instead bang.

Reply:
This actually confused me as well 😄 . I think they are the same. I'm not sure why we're using Next for the null check.

Reply:
@davidfowl do you want to change `segment.Next` null check with `segment.NextSegment` with this PR?

Reply:
I am going to leave it as is and merge this PR, we can revisit later if needed